app/api/reportPost/route.ts

```diff
@@ -0,0 +1,31 @@
+import { NextResponse } from "next/server"
+import db from "@/lib/firebase"
+
+// Function to add a postId to the "reported" collection
+async function reportPost(postId: string) {
+  const reportedRef = db.collection("reported").doc("allReports")
+  const doc = await reportedRef.get()
+  let reportedPosts = []
+
+  if (doc.exists) {
+    reportedPosts = doc.data()?.reportedPosts || []
+  }
+
+  // Append postId if it's not already in the array
+  if (!reportedPosts.includes(postId)) {
```

Comment:
i was thinking maybe a post needs a few reports (maybe 5 or so) to be a "hidden" post, meaning when users see it on the site, its blurred at first and there's an option to view it. after 10+ reports we can consider manually removing it from our frontend. 

i think your solution also works fine for now, but maybe we should include a count so we know how many times a post has been reported. however if we include a count, we will need to only require authenticated users to report.